cartographer/mapping_3d/scan_matching/rotational_scan_matcher.cc

```diff
@@ -122,68 +117,78 @@ sensor::PointCloud SortSlice(const sensor::PointCloud& slice) {
   return result;
 }
 
-std::vector<AngleValuePair> GetValuesForHistogram(
-    const sensor::PointCloud& point_cloud) {
-  std::map<int, sensor::PointCloud> slices;
-  for (const Eigen::Vector3f& point : point_cloud) {
-    slices[common::RoundToInt(point.z() / kSliceHeight)].push_back(point);
+// Rotates the given 'histogram' by the given 'angle'. This might lead to
+// rotations of a fractional bucket which is handled by linearly interpolating.
+Eigen::VectorXf RotateHistogram(const Eigen::VectorXf& histogram,
+                                const float angle) {
+  const float rotate_by_buckets = -angle * histogram.size() / M_PI;
+  int full_buckets = common::RoundToInt(rotate_by_buckets - 0.5f);
+  const float fraction = rotate_by_buckets - full_buckets;
+  while (full_buckets < 0) {
```

Comment:
while -> if and then assert full_buckets > 0?

Reply:
Isn't that just more brittle? I would prefer to make no accuracy assumption here, though I agree that a sane floating point implementation should always be done in one iteration.

Reply:
Sorry, I confused this with the fractional part. This is repeated because the angle might be smaller than -pi.